### PR TITLE
Iemeredu29 683/implement crud operation on object page №1

### PR DESCRIPTION
**Pull request by BabichevDima**

### CODE REVIEW

**Review by TobiaszKowalski — COMMENTED** [review]
- It is better to disable HEAD requests if they have not been implemented in the backend.

image

- Error during the edit description of recently created product

image

- Error during deletion of recently created product
image

File: webapp/i18n/i18n.properties [hunk 1]
```diff
@@ -27,4 +27,6 @@ country=Country
 
 CREATE_DIALOG_TITLE=Create new Category
 
-ST_GENERIC_DELETE_SELECTED=dELLLLEEEETTTEE
\ No newline at end of file
+ST_GENERIC_DELETE_SELECTED=dELLLLEEEETTTEE
```
Comment (TobiaszKowalski) [inline webapp/i18n/i18n.properties]: Strange text. Remove unused i18n texts

File: webapp/ext/controller/ObjectPageExt.controller.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+sap.ui.define([], function () {
+  "use strict";
+  return {
+    onAfterRendering: function () {
+      if (this.oView.byId("project12::sap.suite.ui.generic.template.ObjectPage.view.Details::Categories--ProductID::deleteEntry")) {
+        this.oView.byId("project12::sap.suite.ui.generic.template.ObjectPage.view.Details::Categories--ProductID::deleteEntry").setVisible(false);
+      }
+
+      setTimeout(()=>{
```
Comment (TobiaszKowalski) [inline webapp/ext/controller/ObjectPageExt.controller.js]: Are you sure that setTimeout is a good and only solution?

File: webapp/ext/controller/ObjectPageExt.controller.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        if (this.oView.byId("project12::sap.suite.ui.generic.template.ObjectPage.view.Details::Categories--delete")) {
```
Comment (TobiaszKowalski) [inline webapp/ext/controller/ObjectPageExt.controller.js]: Move id's in a separate file

File: webapp/ext/fragments/Dialog.fragment.xml [hunk 4]
```diff
@@ -0,0 +1,36 @@
+<core:FragmentDefinition xmlns="sap.m"
+	xmlns:f="sap.ui.layout.form"
+	xmlns:core="sap.ui.core">
+	<Dialog title="{i18n>PopupProductTitle}" id="ddd">
```
Comment (TobiaszKowalski) [inline webapp/ext/fragments/Dialog.fragment.xml]: ddd?

File: webapp/ext/controller/ObjectPageExt.controller.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        oODataModel.create(`/${sURL}/Products`, oNewProduct, {
+          success: function () {
+            MessageToast.show("Product was successfully created!");
+          },
+          error: function () {
+            MessageBox.error("Error while creating product!");
+          },
+        });
+
+        // oODataModel.refresh();
```
Comment (TobiaszKowalski) [inline webapp/ext/controller/ObjectPageExt.controller.js]: Remove commented code

File: webapp/ext/controller/ObjectPageExt.controller.js (hunk 2 above)
Comment (BabichevDima) [reply webapp/ext/controller/ObjectPageExt.controller.js]: @TobiaszKowalski it's my better feature. Unfortunately I don't have another idea.

File: webapp/utils/Constants.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+sap.ui.define([], function () {
+  "use strict";
+  return {
```
Comment (TobiaszKowalski) [inline webapp/utils/Constants.js]: You can do "return Object.freeze({//your data})". You don't need function there

File: webapp/ext/controller/ObjectPageExt.controller.js [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+  ["sap/ui/model/json/JSONModel",
+  "sap/m/MessageBox",
+  "sap/m/MessageToast",
+  "project12/utils/Constants"],
+  function (JSONModel, MessageBox, MessageToast, Constants) {
+    "use strict";
+
+    return {
+      onAfterRendering: function () {
+        this.oView.byId(Constants.returnConstant().sDeleteProductButton)?.setVisible(false);
```
Comment (TobiaszKowalski) [inline webapp/ext/controller/ObjectPageExt.controller.js]: ES5

File: webapp/ext/controller/ObjectPageExt.controller.js (hunk 2 above)
Comment (TobiaszKowalski) [reply webapp/ext/controller/ObjectPageExt.controller.js]: @BabichevDima remove setTimeout, remove "?" operator and move condition near the others button checks

**Review by TobiaszKowalski — APPROVED** [review]

**Pull request merged**